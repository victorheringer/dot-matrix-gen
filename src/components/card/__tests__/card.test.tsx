import { render, fireEvent } from "@testing-library/react";
import { Card } from "..";

describe("Card", () => {
  it("should render correctly", () => {
    const tree = render(
      <Card onClick={jest.fn()}>
        <p>test</p>
      </Card>
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should correctly call onPress", () => {
    const mockOnPress = jest.fn();

    const { getByText } = render(
      <Card onClick={mockOnPress}>
        <p>test</p>
      </Card>
    );

    fireEvent.click(getByText("test"));
    expect(mockOnPress).toBeCalledTimes(1);
  });
});
