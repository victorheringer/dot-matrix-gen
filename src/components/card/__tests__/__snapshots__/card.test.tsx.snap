// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ epRxCZ"
  >
    <p>
      test
    </p>
  </div>
</DocumentFragment>
`;
